packages/network-enablement-controller/src/NetworkEnablementController.ts

```diff
@@ -0,0 +1,353 @@
+  }
+
+  getEnabledNetworksForNamespace(namespace: CaipNamespace): string[] {
+    return Object.entries(this.state.enabledNetworkMap[namespace] ?? {})
+      .filter(([, enabled]) => enabled)
+      .map(([id]) => id);
+  }
+
+  getAllEnabledNetworks(): Record<CaipNamespace, string[]> {
+    return (
+      Object.keys(this.state.enabledNetworkMap) as CaipNamespace[]
+    ).reduce(
+      (acc, ns) => {
+        acc[ns] = this.getEnabledNetworksForNamespace(ns);
+        return acc;
+      },
+      {} as Record<CaipNamespace, string[]>,
+    );
+  }
+
+  // ---------------------------- Internals ----------------------------------
+
+  #deriveKeys(chainId: Hex | CaipChainId) {
+    const caipId: CaipChainId = isHexString(chainId)
+      ? toEvmCaipChainId(chainId as Hex)
+      : (chainId as CaipChainId);
+    const { namespace, reference } = parseCaipChainId(caipId);
+    let storageKey: string;
+    if (namespace === (KnownCaipNamespace.Eip155 as string)) {
+      storageKey = isHexString(chainId)
+        ? (chainId as string)
+        : toHex(reference);
+    } else {
+      storageKey = caipId;
+    }
+    return { namespace, storageKey, caipId } as const;
+  }
+
+  #ensureNamespaceBucket(
+    state: NetworkEnablementControllerState,
+    ns: CaipNamespace,
+  ) {
+    if (!state.enabledNetworkMap[ns]) {
+      state.enabledNetworkMap[ns] = {} as Record<string, boolean>;
+    }
+  }
+
+  #ensureNetworkEntry(chainId: Hex | CaipChainId, enable = false): void {
+    const { namespace, storageKey } = this.#deriveKeys(chainId);
+    this.update((s) => {
+      this.#ensureNamespaceBucket(s, namespace);
+      if (!(storageKey in s.enabledNetworkMap[namespace])) {
+        s.enabledNetworkMap[namespace][storageKey] = enable;
+      }
+    });
+  }
+
+  #removeNetworkEntry(chainId: Hex | CaipChainId): void {
+    const { namespace, storageKey } = this.#deriveKeys(chainId);
+    this.update((s) => {
+      delete s.enabledNetworkMap[namespace]?.[storageKey];
+      if (Object.keys(s.enabledNetworkMap[namespace] ?? {}).length === 0) {
+        delete s.enabledNetworkMap[namespace];
+      }
+      this.#ensureAtLeastOneEnabled(s);
+    });
+  }
+
+  #ensureAtLeastOneEnabled(state: NetworkEnablementControllerState) {
+    const anyEnabled = Object.values(state.enabledNetworkMap).some((map) =>
+      Object.values(map).some(Boolean),
+    );
+    if (!anyEnabled) {
+      this.#ensureNamespaceBucket(state, KnownCaipNamespace.Eip155);
+      state.enabledNetworkMap[KnownCaipNamespace.Eip155 as string][
+        ChainId[BuiltInNetworkName.Mainnet]
+      ] = true;
+    }
+  }
+
+  #isKnownNetwork(caipId: CaipChainId): boolean {
+    const { namespace, reference } = parseCaipChainId(caipId);
+    if (namespace === (KnownCaipNamespace.Eip155 as string)) {
+      const { networkConfigurationsByChainId } = this.messagingSystem.call(
+        'NetworkController:getState',
+      );
+      return Boolean(networkConfigurationsByChainId[toHex(reference)]);
```

Comment:
Nit: Instead of using `Boolean`, which could return unexpected results (e.g., `Boolean('')` is `false`), what are your thoughts on testing for presence (which I think is what you're going for here)? It would also make the code more readable.

``` suggestion
      return toHex(reference) in networkConfigurationsByChainId[];